lib/sampler.py

```diff
@@ -101,7 +106,10 @@ def register_process(self, pid, task_id, server_name, worker_id=None,
                 'server_name': server_name,
                 'worker_id': worker_id,
                 'worker_name': worker_name,
+                'rss': get_proc_stat_rss(pid),
             }
+            task = task_id[0] + ((':' + task_id[1]) if task_id[1] else '')
+            self._rss_summary[task] = self.processes[pid]['rss']
```

Comment:
I don't see much value in performing sampling at process registeration. We'll collect RSS at the next sampling point (within 0.1 second) and your measurements (in the commit message) show that it should be often enough to catch a problem. I would remove this code to make it clear how sampling is organized in the code: `_sample_process()` collects per-process metrics, `_sample()` folds them. That's all, ho hacks around.

Reply:
Right, initially I thought the same and tried to avoid of its use:
```
diff --git a/lib/sampler.py b/lib/sampler.py
index 11681e4..a47993c 100644
--- a/lib/sampler.py
+++ b/lib/sampler.py
@@ -108,8 +108,6 @@ class Sampler:
                 'worker_name': worker_name,
                 'rss': get_proc_stat_rss(pid),
             }
-            task = task_id[0] + ((':' + task_id[1]) if task_id[1] else '')
-            self._rss_summary[task] = self._processes[pid]['rss']
             self._log('register', pid)
             return
 
@@ -154,7 +152,7 @@ class Sampler:
                 self._sample_process(pid, tasks_rss)
         # save current overall RSS value if it is bigger than saved
         for task in tasks_rss:
-            if self._rss_summary[task] < tasks_rss[task]:
+            if self._rss_summary and task in self._rss_summary and self._rss_summary[task] < tasks_rss[task]:
                 self._rss_summary[task] = tasks_rss[task]
 
     def _sample_process(self, pid, tasks_rss):
```
but met the issue when there is no any RSS result due to test finished faster than 0.1 sec passed:
```
./test-run.py box/huge_field_map.test.lua 
...
======================================================================================
WORKR TEST                                            PARAMS          RESULT
---------------------------------------------------------------------------------
[001] box/huge_field_map.test.lua                                     [ pass ]
---------------------------------------------------------------------------------
Up to 10 most RSS used tasks in Mb:
-----------------------------------------
Up to 10 most long tasks in seconds:
*   0.03 box/huge_field_map.test.lua                                  
-----------------------------------------
Statistics:
* pass: 1
```
and only when I returned it back needed result was got:
```
./test-run.py box/huge_field_map.test.lua 
...
======================================================================================
WORKR TEST                                            PARAMS          RESULT
---------------------------------------------------------------------------------
[001] box/huge_field_map.test.lua                                     [ pass ]
---------------------------------------------------------------------------------
Up to 10 most RSS used tasks in Mb:
*   67.9 box/huge_field_map.test.lua                                  
-----------------------------------------
Up to 10 most long tasks in seconds:
*   0.03 box/huge_field_map.test.lua                                  
-----------------------------------------
Statistics:
* pass: 1
```

Reply:
After complete offline discussion decided that tests running faster than minimal sample_interval of 0.1 are not interesting in RSS values. Added informational warning to statistics results part:
```
RSS of the failed tasks: (may skip tests run faster than 0.1 sec):
Up to 10 most RSS used tasks in Mb (may skip tests run faster than 0.1 sec):
```